### PR TITLE
intergrate active ocm-lib coding

### PR DESCRIPTION
**Pull request by fabianburth**
**How to categorize this PR?**

/area oci
/kind task
/priority 3

**What this PR does / why we need it**:
This PR is based on #755. 

It provides adapters to implement the introduced facade for working with the ocm with the ocm-lib. Ultimately, this shall enable the landscaper to also work with ocm component descriptor format instead of the legacy cnudie component descriptor format. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature developer
- support for ocm component descriptors
```

### CODE REVIEW

File: pkg/landscaper/blueprints/resolve.go [hunk 1]
```diff
@@ -124,18 +131,25 @@ func Resolve(ctx context.Context, registryAccess model.RegistryAccess, cdRef *ls
...
-	return ResolveBlueprintFromComponentVersion(ctx, componentVersion, bpDef.Reference.ResourceName)
+	resource, err := componentVersion.GetResource(bpDef.Reference.ResourceName, nil)
+	if err != nil {
+		return nil, err
+	}
+	content, err := resource.GetTypedContent(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return content.Resource.(*Blueprint), nil
```
Comment (achimweigel) [inline pkg/landscaper/blueprints/resolve.go:142]: A cast without a check if critical as it will cause a panic stopping the whole system.

File: pkg/landscaper/installations/executions/template/template.go [hunk 2]
```diff
@@ -288,10 +288,7 @@ func serializeComponentDescriptor(componentVersion model.ComponentVersion) (inte
 		return nil, nil
 	}
 
-	cd, err := componentVersion.GetComponentDescriptor()
-	if err != nil {
-		return nil, err
-	}
+	cd := componentVersion.GetComponentDescriptor()
```
Comment (achimweigel) [inline pkg/landscaper/installations/executions/template/template.go:291]: could this return nil?

File: pkg/landscaper/jsonschema/testreg/cert.go [hunk 3]
```diff
@@ -125,7 +125,7 @@ func GenerateCertificates() (*Certificate, error) {
 const chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
 
 func init() {
-	mathrand.Seed(time.Now().Unix())
+	mathrand.New(mathrand.NewSource(time.Now().Unix()))
```
Comment (achimweigel) [inline pkg/landscaper/jsonschema/testreg/cert.go:128]: Why this change? I do not understand the construct at all. The generated Rand object is never used.

**Review by achimweigel — CHANGES_REQUESTED** [review]

File: pkg/landscaper/jsonschema/testreg/cert.go (hunk 3 above)
Comment (fabianburth) [reply pkg/landscaper/jsonschema/testreg/cert.go]: Linter complained after a dependency upgrade that `Seed()` is deprecated. Replaced it with the suggestion provided within the library, but you are right, it is never used. Removed it.

File: pkg/landscaper/blueprints/resolve.go (hunk 1 above)
Comment (fabianburth) [reply pkg/landscaper/blueprints/resolve.go]: I think we can be certain that we have a blueprint at this point. But you are right, better safe than sorry! Added checks to each occurrence.

File: pkg/components/registries/factory.go [hunk 4]
```diff
@@ -1,101 +1,52 @@
...
+	}
 }
 
-func (f *Factory) NewHelmOCIResource(ctx context.Context,
-	ociImageRef string,
-	registryPullSecrets []corev1.Secret,
-	ociConfig *config.OCIConfiguration,
-	sharedCache cache.Cache) (model.Resource, error) {
-	return f.cnudieFactory.NewHelmOCIResource(ctx, ociImageRef, registryPullSecrets, ociConfig, sharedCache)
+func GetFactory() model.Factory {
```
[the remark this answers is not in the document]
Comment (fabianburth) [reply pkg/components/registries/factory.go]: Adjusted. Please check again.

File: pkg/deployer/helm/chartresolver/chartresolver_suite_test.go [hunk 5]
```diff
@@ -61,51 +56,51 @@ var _ = Describe("GetChart", func() {
...
-
-		chartAccess := &helmv1alpha1.Chart{
-			FromResource: ref,
-		}
-
-		chart, err := chartresolver.GetChart(ctx, chartAccess, nil, nil, nil, nil, nil)
-		Expect(err).ToNot(HaveOccurred())
-		Expect(chart.Metadata.Name).To(Equal("ingress-nginx"))
-	})
+	// The team collectively decided that this option to retrieve helm charts is no longer supported
```
[the remark this answers is not in the document]
Comment (fabianburth) [reply pkg/deployer/helm/chartresolver/chartresolver_suite_test.go]: done

**Review by achimweigel — APPROVED** [review]
